hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/replication/ReplicationManager.java

```diff
@@ -425,8 +430,16 @@ public long getReplicationInFlightLimit() {
     // dead. Therefore we simply count the number of healthy nodes and include
     // those which are not in service.
     int healthyNodes = nodeManager.getNodeCount(null, HEALTHY);
+    double inflightOffPeakRatio = isOffPeakHour() ? rmConf.getInflightOffPeakRatio() : 1;
+    LOG.debug("Replication inflight limit, time:{}, inflightOffPeakRatio:{}", LocalDateTime.now(), inflightOffPeakRatio);
     return (long) Math.ceil(healthyNodes * rmConf.getDatanodeReplicationLimit()
-        * factor);
+        * factor * inflightOffPeakRatio);
+  }
+
+  public boolean isOffPeakHour() {
+    List<String> offPeakHours = Arrays.stream(rmConf.getInflightOffPeakHour().split(","))
```

Comment:
We can add a off-peak hour defined or not check before starting split operation.

Reply:
Good idea,I will update the code